javascript/javascript3/week2/currency.js

```diff
@@ -0,0 +1,68 @@
+const amtInput = document.getElementById("amount");
+const fromCurrency = document.getElementById("from-currency");
+const toCurrency = document.getElementById("to-currency");
+const convertedAmount = document.getElementById("converted-amount");
+
+let exchangeRates = {};
+let fromCurrencyValue = "EUR";
+let toCurrencyValue = "DKK";
+
+// Retrieve exchange rates and currencies from the API
+async function logCurrencyRates() {
+  try {
+    const response = await fetch("https://open.er-api.com/v6/latest/USD");
+    const data = await response.json();
+
+    exchangeRates = data.rates;
+    const currencies = Object.keys(exchangeRates);
+
+    const fromOption = document.createElement("option");
+    fromOption.text = "EUR";
+    fromOption.value = "EUR";
+    fromCurrency.appendChild(fromOption);
+
+    const toOption = document.createElement("option");
+    toOption.text = "DKK";
+    toOption.value = "DKK";
+    toCurrency.appendChild(toOption);
+
+    currencies.forEach((currency) => {
+      const option1 = document.createElement("option");
+      option1.text = currency;
+      fromCurrency.add(option1);
+
+      const option2 = document.createElement("option");
+      option2.text = currency;
+      toCurrency.add(option2);
+    });
+  } catch {
+    console.log(error);
+  }
```

Comment:
try to separate the logic here. Async function are normally to do some promises, such as data fetching. So here you can use async function to fetch and store data. then use another function to do the other logics, for example create the DOM elements, doing something with it, etc..

Reply:
Thanks for the feedback